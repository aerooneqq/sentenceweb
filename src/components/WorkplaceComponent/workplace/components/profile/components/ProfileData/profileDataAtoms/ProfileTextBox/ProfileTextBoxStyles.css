.profiletextBoxCont{
  width: 90%;
  margin-left: 5%;
  height: auto;
  background-color: transparent;
}

.propertyValueInput{
  width: 90%;
  height: 30px;

  border-style: solid;
  border-width: 1px;
  border-color: orange;
  border-radius: 5px;

  margin-top: 5px;

  padding-left: 5px;
  padding-right: 5px;
  font-family: Arial;
  font-size: 16px;
}

.propertyDescription{ 
  font-family: Arial;
  font-size: 14px;
  color: gray;

  margin-top: 5px;

  cursor: default;
}

.propertyNameText{
  font-family: Arial;
  font-size: 16px;
  color: black;

  cursor: default;
}

.propertyValueInput:focus{
  outline: none;

  border-color: orangered;
}
