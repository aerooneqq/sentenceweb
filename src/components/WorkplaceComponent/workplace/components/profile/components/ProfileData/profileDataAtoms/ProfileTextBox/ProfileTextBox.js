import React, {Component} from "react"

//Styles
import "./ProfileTextBoxStyles.css"

/**
 * PROPS LIST:
 * 1) updateData - when the content of the text box is changed it updates the corresponding
 *                 property in the user object.
 * 2) propertyName
 * 3) propertyDescription
 * 4) propertyValue - the initial value of the property
 */
export default class ProfileTextBox extends Component{

  constructor(props) {
    super(props);

    this.state={
      value: props.propertyValue
    };

    this.onInputChange = this.onInputChange.bind(this);
  }

  onInputChange(event) {
    let modelPropertyName = this.props.propertyName;
    modelPropertyName = modelPropertyName.charAt(0).toLowerCase() + modelPropertyName.substring(1);

    this.props.updateData(modelPropertyName, event.target.value);

    this.setState({value: event.target.value});
  }

  render() {
    return (
      <div className = "profiletextBoxCont">
        <div className = "propertyNameText">{this.props.propertyName}</div>
        <div className = "propertyDescription">{this.props.propertyDescription}</div>
        <input className = "propertyValueInput" type = "text" value={this.state.value} onChange={this.onInputChange} />
      </div>
    );
  }
}
