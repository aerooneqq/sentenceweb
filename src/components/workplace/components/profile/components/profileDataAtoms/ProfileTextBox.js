import React, {Component} from "react"
import "./styles/ProfileTextBoxStyles.css"

export default class TextBox extends Component{
  constructor(props){
    super(props);

    this.state={
      value: props.propertyValue
    };

    this.onInputChange = this.onInputChange.bind(this);
  }

  onInputChange(event){
    this.setState({
      value: event.target.value
    });
  }

  render(){
    return(
      <div className = "profiletextBoxCont">
        <div className = "propertyNameText">{this.props.propertyName}</div>
        <div className = "propertyDescription">{this.props.propertyDescription}</div>
        <input className = "propertyValueInput" type = "text" value={this.state.value} onChange={this.onInputChange}></input>
      </div>
    )
  }
}
