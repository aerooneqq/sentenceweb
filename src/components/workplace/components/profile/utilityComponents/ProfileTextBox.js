import React, {Component} from "react"
import "../styles/ProfileTextBoxStyles.css"

export default class TextBox extends Component{
  constructor(props){
    super(props)
  }

  render(){
    return(
      <div className = "profiletextBoxCont">
        <p className = "propertyNameText">Property name</p>
        <input className = "propertyValueInput" type = "text" value="Property value"></input>
      </div>
    )
  }
}
