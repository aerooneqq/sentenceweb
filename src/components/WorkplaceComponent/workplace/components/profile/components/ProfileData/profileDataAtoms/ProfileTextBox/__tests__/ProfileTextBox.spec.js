import React from "react";
import {shallow} from "enzyme"; 
import {shallowToJson} from "enzyme-to-json";
import ProfileTextBox from "../ProfileTextBox";

const spy = jest.fn(() => {});
const props = { 
    propertyName: "property name",
    propertyDescription: "property description",
    propertyValue: "value",
    updateData: spy
}

describe("ProfileTextBox", () => { 
    it("should render correctly", () => { 
        expect(shallowToJson(shallow(<ProfileTextBox {...props}/>))).toMatchSnapshot();
    });

    it("should have proper intiial state", () => { 
        const profileTextBox = shallow(<ProfileTextBox {...props}/>);

        const properState = { 
            value: props.propertyValue
        }

        for (let key in properState) { 
            expect(properState[key]).toBe(profileTextBox.state(key));
        }
    });

    it("should call updateData from props when content is changed", () => { 
        const profileTextBox = shallow(<ProfileTextBox {...props}/>);

        const changeEvent = {
            preventDefault() {},
            target: { value: 'the-value' }
        }

        profileTextBox.find("input.propertyValueInput").simulate("Change", changeEvent);

        expect(spy).toBeCalled();
        expect(profileTextBox.state("value")).toBe(changeEvent.target.value);
    });

    it("should display right propert name and property description", () => { 
        const profileTextBox = shallow(<ProfileTextBox {...props}/>);
    
        expect(profileTextBox.find("div.propertyNameText").text()).toBe(props.propertyName);
        expect(profileTextBox.find("div.propertyDescription").text()).toBe(props.propertyDescription);
    });
});