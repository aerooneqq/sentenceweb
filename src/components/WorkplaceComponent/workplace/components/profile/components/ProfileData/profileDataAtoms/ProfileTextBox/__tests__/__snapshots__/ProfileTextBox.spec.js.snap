// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileTextBox should render correctly 1`] = `
<div
  className="profiletextBoxCont"
>
  <div
    className="propertyNameText"
  >
    property name
  </div>
  <div
    className="propertyDescription"
  >
    property description
  </div>
  <input
    className="propertyValueInput"
    onChange={[Function]}
    type="text"
    value="value"
  />
</div>
`;
